plutus-tx/src/PlutusTx/Foldable.hs

```diff
@@ -225,6 +228,20 @@ find p = coerce . foldMap (\ x -> First (if p x then Just x else Nothing))
 -- | Plutus Tx version of 'Data.Foldable.mapM_'.
 {-# INLINABLE mapM_ #-}
 mapM_ :: (Foldable t, Haskell.Monad m) => (a -> m b) -> t a -> m ()
+-- See Note [Haskell.Monad]
 mapM_ f = foldr c (Haskell.return ())
   where c x k = f x Haskell.>> k
         {-# INLINE c #-}
+
+{- Note [Haskell.Monad]
+There is no `PlutusTx.Monad`. Since do-notations use `Haskell.Monad` (without
+`RebindableSyntax` anyway), it won't be very useful to have `PlutusTx.Monad`.
```

Comment:
Maybe we should just suggest that people use `RebindableSyntax` if they want to use do notation? It's pretty sketchy what we're doing right now.

Reply:
I think this note just makes us sound like we know what we're doing too much :D I think the truth is probably more like "we didn't want to bother with dealing with `QualifiedDo` etc. (and it's only in GHC 9), and the status quo seemed to work so we didn't look too hard".